app/components/davros-uploader.js

```diff
@@ -46,7 +46,7 @@ export default class DavrosUploader extends Component {
 
     var fullPath = ['/', this.uploadLocation, path].join('');
 
-    yield File.ensureCollectionExists(fullPath).then(() => {
+    yield File.ensureCollectionExists(decodeURI(fullPath)).then(() => {
```

Comment:
Rather than encoding and then decoding, can we defer the encoding step or do it on a parallel path? (I'm actually not even sure why URI-encoding is happening in the first place.)